app/controllers/immunisation_imports/patients_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ImmunisationImports::PatientsController < ApplicationController
+  before_action :set_campaign
+  before_action :set_immunisation_import
+  before_action :set_patient
+
+  def show
+    render layout: "full"
+  end
+
+  def update
+  end
+
+  private
+
+  def set_campaign
+    @campaign =
+      policy_scope(Campaign)
+        .active
+        .includes(:immunisation_imports)
+        .find(params[:campaign_id])
+  end
+
+  def set_immunisation_import
+    @immunisation_import =
+      @campaign.immunisation_imports.find(params[:immunisation_import_id])
+  end
+
+  def set_patient
+    @patient =
+      @immunisation_import
+        .vaccination_records
+        .map(&:patient)
+        .find(params[:id])
+        .first
```

Comment:
Do we need the `.first` here? I'd expect `find` to return a single object.

Reply:
It returned an enumerator initially but I'll look at this code again. I'm also not pleased that I have to `map(&:patients)` and iterate in Ruby instead of AR, so I'd like to golf this to something else.

Reply:
I've updated this with an AR-based version; it doesn't look too great, but it's faster